src/TwigComponent/src/ComponentTemplateFinder.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Twig\Environment;
+
+/**
+ * @author Matheo Daninos <matheo.daninos@gmail.com>
+ */
+final class ComponentTemplateFinder implements ComponentTemplateFinderInterface
```

Comment:
This feels a bit like a premature abstraction, what about in-lining in `ComponentFactory` for now?

Reply:
@weaverryan ask me to do it in this comment: https://github.com/symfony/ux/pull/802#discussion_r1253396681.
To not have to autowire the Environment service